pkg/virt-controller/watch/vmi.go

```diff
@@ -709,6 +712,16 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 
 	case vmi.IsRunning():
 		if !vmiPodExists {
+			virtHandlerAlive, err := isVirtHandlerAlive(c.clientset, vmi.Status.NodeName)
+
+			if err != nil {
+				log.Log.Errorf("virt-handler pod is not responsible %v", err)
+			}
+
+			if !virtHandlerAlive && (!vmiPodExists || isPodDownOrGoingDown(pod)) {
+				vmiCopy.Status.Phase = virtv1.Failed
+			}
```

Comment:
the`&& (!vmiPodExists || isPodDownOrGoingDown(pod)` part of this isn't necessary or even accurate. The pod is most likely `nil` here which would cause `isPodDownOrGoingDown` to segfault.

you're already in an if clause that's checking if `!vmiPodExists` and there shouldn't be a reason to look at `pod`

Reply:
Done.